server/src/github/middleware.rs

```diff
@@ -0,0 +1,90 @@
+use std::future::{ready, Ready};
+
+
+use actix_web::{
+    dev::{forward_ready, Service, ServiceRequest, ServiceResponse, Transform, Payload},
+    Error,
+};
+use futures_util::future::LocalBoxFuture;
+
+// There are two steps in middleware processing.
+// 1. Middleware initialization, middleware factory gets called with
+//    next service in chain as parameter.
+// 2. Middleware's call method gets called with normal request.
+pub struct VerifyGithubSignatureFactory;
+
+// Middleware factory is `Transform` trait
+// `S` - type of the next service
+// `B` - type of response's body
+impl<S, B> Transform<S, ServiceRequest> for VerifyGithubSignatureFactory
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = VerifyGithubSignatureMiddleware<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        ready(Ok(VerifyGithubSignatureMiddleware { service }))
+    }
+}
+
+pub struct VerifyGithubSignatureMiddleware<S> {
+    service: S,
+}
+
+impl<S, B> Service<ServiceRequest> for VerifyGithubSignatureMiddleware<S>
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
```

Comment:
B is an interesting choice for generic name - I can't actually think what it would be other than just a random placeholder letter. S makes sense as its the Service type (I assume), B not so much. I am not experienced with this actix middleware though so it might be framework specific...

Reply:
Let me dig into it more, it's a good question, tbh it's just what was in their example and I don't really know why they picked B myself.